internal/resolver/delegatingresolver/delegatingresolver_test.go

```diff
@@ -0,0 +1,319 @@
+}
+
+// TestDelegatingResolverNoProxy verifies the behavior of the delegating resolver
+// when no proxy is configured.
+func (s) TestDelegatingResolverNoProxy(t *testing.T) {
+	// Enable HTTP Proxy env var.
+	defer overwriteAndRestoreProxyEnv("")()
+	mr := manual.NewBuilderWithScheme("test") // Set up a manual resolver to control the address resolution.
+	target := "test:///" + targetTestAddr
+
+	tcc, stateCh, _ := createTestResolverClientConn(t)
+	// Create a delegating resolver with no proxy configuration
+	dr, err := New(resolver.Target{URL: *testutils.MustParseURL(target)}, tcc, resolver.BuildOptions{}, mr, false)
+	if err != nil || dr == nil {
+		t.Fatalf("Failed to create delegating resolver: %v", err)
+	}
+
+	// Update the manual resolver with a test address.
+	mr.UpdateState(resolver.State{
+		Addresses:     []resolver.Address{{Addr: resolvedTargetTestAddr}},
+		ServiceConfig: &serviceconfig.ParseResult{},
+	})
+
+	// Verify that the delegating resolver outputs the same address.
+	expectedState := resolver.State{
+		Addresses:     []resolver.Address{{Addr: resolvedTargetTestAddr}},
+		ServiceConfig: &serviceconfig.ParseResult{},
+	}
+	state := <-stateCh
+	if len(state.Addresses) != 1 || !cmp.Equal(expectedState, state) {
+		t.Errorf("Unexpected state from delegating resolver: %v, want %v", state, expectedState)
+	}
+}
+
+// setupDNS unregisters the DNS resolver and registers a manual resolver for the
+// same scheme. This allows the test to mock the DNS resolution for the proxy resolver.
+func setupDNS(t *testing.T) *manual.Resolver {
+
+	mr := manual.NewBuilderWithScheme("dns")
+
+	dnsResolverBuilder := resolver.Get("dns")
+	resolver.Register(mr)
+
+	t.Cleanup(func() { resolver.Register(dnsResolverBuilder) })
+	return mr
+}
+
+// TestDelegatingResolverwithDNSAndProxy verifies that the delegating resolver
+// correctly updates state when the target URI scheme is DNS and a proxy is
+// configured and target resolution is enabled.
+func (s) TestDelegatingResolverwithDNSAndProxyWithTargetResolution(t *testing.T) {
+	// Enable HTTP Proxy env var.
+	defer overwriteAndRestoreProxyEnv(envProxyAddr)()
+	hpfe := func(req *http.Request) (*url.URL, error) {
+		if req.URL.Host == targetTestAddr {
+			return &url.URL{
+				Scheme: "https",
+				Host:   envProxyAddr,
+			}, nil
+		}
+		return nil, nil
+	}
+	defer overwrite(hpfe)()
+	mrTarget := setupDNS(t) // Manual resolver to control the target resolution.
+	mrProxy := setupDNS(t)  // Set up a manual DNS resolver to control the proxy address resolution.
+	target := "dns:///" + targetTestAddr
+
+	tcc, stateCh, _ := createTestResolverClientConn(t)
+	dr, err := New(resolver.Target{URL: *testutils.MustParseURL(target)}, tcc, resolver.BuildOptions{}, mrTarget, true)
+	if err != nil {
+		t.Fatalf("Failed to create delegating resolver: %v", err)
+	}
+	if dr == nil {
+		t.Fatalf("Failed to create delegating resolver")
+	}
+	mrTarget.UpdateState(resolver.State{
```

Comment:
same here, even before updating we should verify if the initial addresses are correct